sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -693,6 +719,12 @@ public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
 
     @Override
     public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offersToProcess) {
+        if (!apiServerDisabled.get() && (apiServer == null || !apiServer.isStarted())) {
+            LOGGER.info("Waiting for API Server to start ...");
+            declineOffers(driver, Collections.emptyList(), offersToProcess);
+            return;
```

Comment:
It is relevant.  I'm going to leave a timeout out of this PR.  @nickbp. that 60s timeout was for the Scheduler to provide in memory Resource objects.  That timeout was NEVER hit.  It was an artifact of the weird interdependent design with that weird queue.  So not relevant anymore.  I am hesitant to put a timeout now given ES behavior.  I don't want the Scheduler crash looping because it's on a slow machine.  Happy to revisit if we see this screwing things up.